CHANGELOG.md

```diff
@@ -1,7 +1,9 @@
 # Karafka framework changelog
 
 ## 2.2.14 (Unreleased)
+- [Change] Replace single `#before_schedule` with appropriate methods and events for scheduling various types of work. This is needed as we first of all may run different framework logic on those and second for accurate jobs tracking with advanced schedulers.
```

Comment:
Replace single `#before_schedule` with appropriate methods and events for scheduling various types of work. This is needed as we may run different framework logic on those and, second, for accurate job tracking with advanced schedulers.